assets/src/widget/stories-widget.js

```diff
@@ -40,18 +40,34 @@ const reactiveWidget = function (target, reset = false) {
   for (const [key, value] of Object.entries(state)) {
     const field = widget.querySelector(`.${key}.stories-widget-field`);
     const fieldWrapper = widget.querySelector(`.${key}_wrapper`);
-    if (field && fieldWrapper && 'checkbox' === field.getAttribute('type')) {
-      if (reset) {
-        field.checked = false;
-      }
-      /**
-       * If this is readonly field.
-       * Assign the value automatically and hide it afterward.
-       */
-      if (value.readonly) {
-        field.checked = value.show;
+
+    if (field && fieldWrapper) {
+      const field_type = field.getAttribute('type');
+
+      switch (field_type) {
```

Comment:
Case statement with a single option and default seems a little silly here. Wouldn't an if else do as well?

Reply:
@spacedmonkey  We can have some more control types in future based on view type, it will make sense to use switch there.

Reply:
Maybe switch an if / else with a todo to use a switch statement if more options are added? How likely is it that new view types will be added?